lib/mongoid/reloadable.rb

```diff
@@ -82,7 +82,7 @@ def reload_embedded_document
     def extract_embedded_attributes(attributes)
       atomic_position.split(".").inject(attributes) do |attrs, part|
         attrs = attrs[part =~ /\d/ ? part.to_i : part]
-        attrs
+        attrs || {}
```

Comment:
I don't think this is correct. If there are not attributes at the specified path (`atomic_position`), this method should probably return nil and the docstring should be updated to reflect that.

Reply:
okay that was my original fix, I'll go back to that